src/lxc/conf.c

```diff
@@ -4092,6 +4092,24 @@ static int send_ttys_to_parent(struct lxc_handler *handler)
 	return -1;
 }
 
+int lxc_early_setup(struct lxc_handler *handler)
+{
+	const char *name = handler->name;
+	struct lxc_conf *lxc_conf = handler->conf;
+
+	if (lxc_create_tty(name, lxc_conf)) {
+		ERROR("failed to create the ttys");
+		return -1;
+	}
+
+	if (send_ttys_to_parent(handler) < 0) {
+		ERROR("failure sending console info to parent");
+		return -1;
+	}
```

Comment:
So what is the reasoning for putting the tty creation and sending before unsharing the cgroup namespace?